src/main/java/model/FigureSupplier.java

```diff
@@ -1,55 +1,45 @@
 package model;
 
+import java.util.Random;
+
 public class FigureSupplier {
     public Figure getRandomFigure() {
-        //creating parametr which will be used to choose figure
-        int random = (int)(Math.random() * 100);
-        //creating an array with demensions of figure
-        double a = Math.random();
-        Long randomlong = (long)Math.round(a * 1000000000) + 1111111;
-        char[] arrch = Long.toString(randomlong).toCharArray();
-        String v0 = Character.toString(arrch[0]) + Character.toString(arrch[1]);
-        String v1 = Character.toString(arrch[2]) + Character.toString(arrch[3]);
-        String v2 = Character.toString(arrch[4]) + Character.toString(arrch[5]);
+        Random random = new Random();
+        //creating an array with dimensions of figure
         int[] finalInt = new int[3];
-        finalInt[0] = Integer.valueOf(v0) + 1;
-        finalInt[1] = Integer.valueOf(v1) + 1;
-        finalInt[2] = Integer.valueOf(v2) + 1;
+        finalInt[0] = random.nextInt(101) + 1;
+        finalInt[1] = random.nextInt(101) + 1;
+        finalInt[2] = random.nextInt(101) + 1;
         //choosing the figure
-        switch (random % 5) {
+        switch (random.nextInt() % 5) {
             case (0):
                 // creating circle
                 Circle circle = new Circle(ColorSupplier.getRandomColor());
-                circle.setColour(ColorSupplier.getRandomColor());
-                circle.setParams(finalInt);
+                circle.setParams(finalInt, ColorSupplier.getRandomColor());
```

Comment:
let's create private methods for figure creation. also, color is being set twice here